File: app/src/main/java/com/lihang/mysmartloadingview/MainActivity.java
package com.lihang.mysmartloadingview;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import com.gyf.barlibrary.ImmersionBar;
import com.lihang.mysmartloadingview.databinding.ActivityMainBinding;
import com.lihang.SmartLoadingView;

import java.util.concurrent.TimeUnit;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;


public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    //沉浸式状态栏
    protected ImmersionBar mImmersionBar;
    ActivityMainBinding binding;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.setOnClickListener(this);
        mImmersionBar = ImmersionBar.with(this);
        mImmersionBar.init();

    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.smartLoadingView_1_:
                binding.smartLoadingView1.start();
                Observable.timer(2000, TimeUnit.MILLISECONDS)
                        .observeOn(AndroidSchedulers.mainThread()).subscribe(along -> {
                    binding.smartLoadingView1.onSuccess(MainActivity.this, SecondActivity.class);
                });
                break;

            case R.id.smartLoadingView_2_:
                binding.smartLoadingView2.start();
                Observable.timer(2000, TimeUnit.MILLISECONDS)
                        .observeOn(AndroidSchedulers.mainThread()).subscribe(along -> {
                    binding.smartLoadingView2.onSuccess(new SmartLoadingView.AnimationFullScreenListener() {
                        @Override
                        public void animationFullScreenFinish() {
                            Toast.makeText(MainActivity.this, "监听动画结束", Toast.LENGTH_SHORT).show();
                        }
                    });
                });
                break;

            case R.id.smartLoadingView_3_:
                binding.smartLoadingView3.start();
                Observable.timer(2000, TimeUnit.MILLISECONDS)
                        .observeOn(AndroidSchedulers.mainThread()).subscribe(along -> {
                    binding.smartLoadingView3.netFaile();
                });
                break;
            case R.id.smartLoadingView_4_:


                if (binding.smartLoadingView4.isFollow()) {
                    //这里是模拟取消关注
                    binding.smartLoadingView4.reset();
                } else {
                    //这里是模拟关注
                    binding.smartLoadingView4.start();
                    Observable.timer(2000, TimeUnit.MILLISECONDS)
                            .observeOn(AndroidSchedulers.mainThread()).subscribe(along -> {
                        binding.smartLoadingView4.netFaile("关注成功");
                    });
                }

                break;


            case R.id.smartLoadingView_5_:
                //不设置模式，那么默认为正常模式 SmartLoadingView.OKAnimationType.NORMAL

                if (binding.smartLoadingView5.isFollow()) {
                    binding.smartLoadingView5.reset();
                } else {
                    binding.smartLoadingView5.start();
                    Observable.timer(2000, TimeUnit.MILLISECONDS)
                            .observeOn(AndroidSchedulers.mainThread()).subscribe(along -> {
                        binding.smartLoadingView5.onSuccess(new SmartLoadingView.AnimationOKListener() {
                            @Override
                            public void animationOKFinish() {
                                Log.e("我这里真的走了两次", " --------  ");
                                Toast.makeText(MainActivity.this, "关注成功", Toast.LENGTH_SHORT).show();
                            }
                        });
                    });
                }


                break;


            case R.id.smartLoadingView_hide:
                binding.smartLoadingViewHide.start();
                Observable.timer(2000, TimeUnit.MILLISECONDS)
                        .observeOn(AndroidSchedulers.mainThread()).subscribe(along -> {
                    binding.smartLoadingViewHide.onSuccess(new SmartLoadingView.AnimationOKListener() {
                        @Override
                        public void animationOKFinish() {
                            Toast.makeText(MainActivity.this, "关注成功", Toast.LENGTH_SHORT).show();
                        }
                    });
                });
                break;

            case R.id.smartLoadingView_center:
                binding.smartLoadingViewCenter.start();
                Observable.timer(2000, TimeUnit.MILLISECONDS)
                        .observeOn(AndroidSchedulers.mainThread()).subscribe(along -> {
                    binding.smartLoadingViewCenter.onSuccess(new SmartLoadingView.AnimationOKListener() {
                        @Override
                        public void animationOKFinish() {
                            Toast.makeText(MainActivity.this, "关注成功", Toast.LENGTH_SHORT).show();
                        }
                    });
                });
                break;

            case R.id.smartLoadingView_6_:
                binding.smartLoadingView6.start();
                Observable.timer(2000, TimeUnit.MILLISECONDS)
                        .observeOn(AndroidSchedulers.mainThread()).subscribe(along -> {
                    Toast.makeText(MainActivity.this, "关注失败，回到开始", Toast.LENGTH_SHORT).show();
                    binding.smartLoadingView6.backToStart();
                });
                break;

            case R.id.smartLoadingView_7_:
                binding.smartLoadingView7.start();
                Observable.timer(2000, TimeUnit.MILLISECONDS)
                        .observeOn(AndroidSchedulers.mainThread()).subscribe(along -> {
                    binding.smartLoadingView7.netFaile();
                });
                break;
            case R.id.smartLoadingView_8_:
                binding.smartLoadingView8.start();
                Observable.timer(2000, TimeUnit.MILLISECONDS)
                        .observeOn(AndroidSchedulers.mainThread()).subscribe(along -> {
                    binding.smartLoadingView8.netFaile();
                });
                break;

            case R.id.smartLoadingView_9_:
                Toast.makeText(MainActivity.this, "点击了", Toast.LENGTH_SHORT).show();
                binding.smartLoadingView9.setEnabled(false);
                break;

            case R.id.smartLoadingView_login_demo:
                binding.smartLoadingViewLoginDemo.start();
                Observable.timer(2000, TimeUnit.MILLISECONDS)
                        .observeOn(AndroidSchedulers.mainThread()).subscribe(along -> {
                    binding.smartLoadingViewLoginDemo.onSuccess(new SmartLoadingView.AnimationFullScreenListener() {
                        @Override
                        public void animationFullScreenFinish() {
                            startActivity(new Intent(MainActivity.this, LoginActivity.class));
                            overridePendingTransition(R.anim.scale_test_home, R.anim.scale_test2);
                        }
                    });
                });
                break;
            case R.id.smartLoadingView_follow_demo:
                binding.smartLoadingViewFollowDemo.start();
                Observable.timer(2000, TimeUnit.MILLISECONDS)
                        .observeOn(AndroidSchedulers.mainThread()).subscribe(along -> {
                    binding.smartLoadingViewFollowDemo.onSuccess(new SmartLoadingView.AnimationFullScreenListener() {
                        @Override
                        public void animationFullScreenFinish() {
                            startActivity(new Intent(MainActivity.this, FollowActivity.class));
                            overridePendingTransition(R.anim.scale_test_home, R.anim.scale_test2);
                        }
                    });
                });
                break;
        }

    }


    @Override
    protected void onResume() {
        super.onResume();

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mImmersionBar.destroy();
    }


}

File: app/src/main/java/com/lihang/mysmartloadingview/MainActivity.kt
package com.lihang.mysmartloadingview

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Gravity
import android.view.View
import android.widget.Toast
import com.blankj.utilcode.util.ActivityUtils
import com.blankj.utilcode.util.ToastUtils
import com.lihang.SmartLoadingView
import com.lihang.SmartLoadingView.LoadingListener
import com.lihang.mysmartloadingview.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var mBinding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        supportActionBar?.title = "SmartLoadingView的使用"
        ToastUtils.setGravity(Gravity.CENTER,0,0)
        //
        mBinding.run {
            /**
             * 1.smart_full_screen:全屏
             * 不支持关注模式
             * */

            //finishLoadingWithFullScreen 的使用
            smartFullscreenAuto.setOnClickListener {
                //1.1、开始加载
                smartFullscreenAuto.startLoading()

                //1.2、模拟2s后，加载成功并跳转（finishLoadingWithFullScreen会自动跳转页面并关闭当前页面）
                it.postDelayed({
                    smartFullscreenAuto.finishLoadingWithFullScreen(this@MainActivity, SecondActivity::class.java)
                    //也可以自己处理动画回调
                    //smartFullscreenAuto.finishLoading(true) {}
                }, 2000)
            }

            //模拟失败
            smartFullscreenFail.setOnClickListener {
                smartFullscreenFail.startLoading()
                it.postDelayed({
                    smartFullscreenFail.finishLoading(false) {
                        ToastUtils.showShort("加载失败")
                    }
                }, 2000)
            }


            //模拟失败，并展示失败文案
            smartFullscreenFailtxt.setOnClickListener {
                smartFullscreenFailtxt.startLoading()
                it.postDelayed({
                    //smartFullscreenFailtxt.setAnimaledText("我是自定义错误文案")
                    smartFullscreenFailtxt.finishLoading(false)
                }, 2000)
            }


            /**
             * 2.smart_button
             * 支持关注模式
             * */
            //正常情况下的："点击关注" 和 "取消关注"
            smartButtonSuccess.setOnClickListener {
                if (!smartButtonSuccess.isFinished) {
                    smartButtonSuccess.startLoading()
                    it.postDelayed({
                        //2.1、加载成功--带loading动画的加载成功
                        smartButtonSuccess.finishLoading(true) {
                            ToastUtils.showShort("关注成功")
                        }
                    }, 2000)
                } else {
                    //2.2、再次点击，取消关注。通过此方法设置不带动画
                    smartButtonSuccess.isFinished = false
                }
            }

            //模拟关注失败
            smartButtonFail.setOnClickListener {
                smartButtonFail.startLoading()
                it.postDelayed({
                    smartButtonFail.finishLoading(false) {
                        ToastUtils.showShort("关注失败，请稍后再试~")
                    }
                }, 2000)
            }


            /**
             * 3.smart_tick
             * 支持关注模式
             * */
            //这里的用法和 smart_button 类似
            smartTickDemo.setOnClickListener {
                if (!smartTickDemo.isFinished) {
                    smartTickDemo.startLoading()
                    it.postDelayed({
                        //2.1、加载成功--带loading动画的加载成功
                        smartTickDemo.finishLoading(true) {
                            ToastUtils.showShort("关注成功")
                        }
                    }, 2000)
                } else {
                    //2.2、再次点击，取消关注。通过此方法设置不带动画
                    smartTickDemo.isFinished = false
                }
            }


            /**
             * 4.smart_tick_hide
             * 支持关注模式
             * */
            //这里的用法和 smart_button 类似
            smartTickHideDemo.setOnClickListener {
                if (!smartTickHideDemo.isFinished) {
                    smartTickHideDemo.startLoading()
                    it.postDelayed({
                        //2.1、加载成功--带loading动画的加载成功
                        smartTickHideDemo.finishLoading(true) {
                            ToastUtils.showShort("关注成功")
                        }
                    }, 2000)
                } else {
                    //2.2、再次点击，取消关注。通过此方法设置不带动画
                    smartTickHideDemo.isFinished = false
                }
            }


            /**
             * 5.smart_tick_center_hide
             * 支持关注模式
             * */
            //这里的用法和 smart_button 类似
            smartTickCenterHideDemo.setOnClickListener {
                if (!smartTickCenterHideDemo.isFinished) {
                    smartTickCenterHideDemo.startLoading()
                    it.postDelayed({
                        //2.1、加载成功--带loading动画的加载成功
                        smartTickCenterHideDemo.finishLoading(true) {
                            ToastUtils.showShort("关注成功")
                        }
                    }, 2000)
                } else {
                    //2.2、再次点击，取消关注。通过此方法设置不带动画
                    smartTickCenterHideDemo.isFinished = false
                }
            }

        }
    }
}